files/pull-sandbox-image.sh

```diff
@@ -1,16 +1,25 @@
 #!/usr/bin/env bash
-set -euo pipefail
 
 source <(grep "sandbox_image" /etc/containerd/config.toml | tr -d ' ')
 
+### skip if we don't have a sandbox_image set in config.toml
+if [[ -z ${sandbox_image:-} ]]; then
+  echo >&2 "Skipping ... missing sandbox_image from /etc/containerd/config.toml"
+  exit 0
+fi
+
 ### Short-circuit fetching sandbox image if its already present
-if [[ "$(sudo ctr --namespace k8s.io image ls | grep $sandbox_image)" != "" ]]; then
+if [[ -n $(sudo ctr --namespace k8s.io image ls | grep "${sandbox_image}") ]]; then
+  echo >&2 "Skipping ... sandbox_image '${sandbox_image}' is already present"
   exit 0
 fi
 
-# use the region that the sandbox image comes from for the ecr authentication,
-# also mitigating the localzone isse: https://github.com/aws/aws-cli/issues/7043
-region=$(echo "${sandbox_image}" | cut -f4 -d ".")
+# if the sandbox image is provided by the bootstrap script, then the region is
+# guaranteed to come from this data source.
+# see: https://github.com/awslabs/amazon-eks-ami/blob/baef6f0860f60dbec366de30853e47418e3fb430/files/bootstrap.sh#L320-L338
+# if the image is customer provided, then this is just a sane default for the
+# region when attempting to get ecr credentials.
+region=$(imds 'latest/dynamic/instance-identity/document' | jq .region -r)
```

Comment:
one problem here is ... if the sandbox_image is specified from another region ( different from where the node is running in :( )

Reply:
could we check if the echo/cut version (from the sandbox_image) is present in `aws account list-regions`? and skip if not?

Reply:
oh but this would require new permissions on the role 😖

Reply:
are there docs stating required naming conventions of ecr repos? I'm not 100% sure what i can/can't check for

Reply:
> if the sandbox_image is specified from another region

but this is only for user provided images right (based on the assumptions for the eks generated uri), which I'm not sure had a solution even prior to this 🤔

Reply:
then we can let it slide and leave it to the `imds` variation. (cross fingers!)

Reply:
I'm hesitant to touch this in this PR. Cutting the region out of the image reference has it's caveats, for sure; but that logic has been in prod for over a year and it's proven solid. I think we should consider this change in a different PR, since this one is mostly for mitigation

Reply:
> that logic has been in prod for over a year and it's proven solid.

This logic has existed yes (https://github.com/awslabs/amazon-eks-ami/blame/master/files/pull-image.sh#L4), but its only been used in context where the image source was already known and decided by us right?

The only prior usage i know if is in
https://github.com/awslabs/amazon-eks-ami/blob/baef6f0860f60dbec366de30853e47418e3fb430/scripts/install-worker.sh#L484

Reply:
`pull-sandbox-image.sh` used to call `pull-image.sh` instead of using `crictl`

Reply:
🤦 that escaped my mind (and my github searches)
yea i think pulling it out makes sense then since it won't break the expectation beyond whats already changed in last commit